setup.sh

```diff
@@ -247,39 +247,50 @@ EOF
 )
 
 setup_osie() (
-	mkdir -p "$STATEDIR/webroot"
-
 	local osie_current=$STATEDIR/webroot/misc/osie/current
 	local tink_workflow=$STATEDIR/webroot/workflow/
-	if [ ! -d "$osie_current" ] || [ ! -d "$tink_workflow" ]; then
-		mkdir -p "$osie_current"
-		mkdir -p "$tink_workflow"
-		pushd "$SCRATCH"
-
-		if [ -z "${TB_OSIE_TAR:-}" ]; then
-			local OSIE_DOWNLOAD_LINK=${OSIE_DOWNLOAD_LINK:-"https://tinkerbell-oss.s3.amazonaws.com/osie-uploads/latest.tar.gz"} # If variable not set or null, use default.
-			curl -fsSL "${OSIE_DOWNLOAD_LINK}" -o ./osie.tar.gz
-			tar -zxf osie.tar.gz
-		else
-			if [ ! -f "$TB_OSIE_TAR" ]; then
-				echo "$ERR osie tar not found in the given location $TB_OSIE_TAR"
-				exit 1
-			fi
-			echo "$INFO extracting osie tar"
-			tar -zxf "$TB_OSIE_TAR"
+
+	if [ -d "$osie_current" ] && [ -d "$tink_workflow" ]; then
```

Comment:
this should probably check for 

```suggestion
	if [ -f "$osie_current/vmlinuz-x86_64" ] && [ -f "$tink_workflow/workflow-helper.sh" ]; then
```

instead. Otherwise if curl fails then a re-run will skip curl right? Because you're `mkdir -p` these dirs right after this branch